src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenLocalFunctionTests.cs

```diff
@@ -30,6 +30,114 @@ public static IMethodSymbol FindLocalFunction(this CommonTestBase.CompilationVer
     public class CodeGenLocalFunctionTests : CSharpTestBase
     {
         [Fact]
+        public void VS15Pre3BugTest()
+        {
+            var src = @"
+using System;
+using System.Collections.Generic;
+
+class Test<T>
+{
+    T Value;
+
+    public bool Foo(IEqualityComparer<T> comparer)
+    {
+        bool local(T tmp)
+        {
+            return comparer.Equals(tmp, this.Value);
+        }
+        return local(this.Value);
+    }
+}
+";
+            var comp = CompileAndVerify(src);
+        }
+
+        [Fact]
+        public void CaptureGenericField()
+        {
+            var src = @"
+using System;
+class C<T>
+{
+    T Value = default(T);
+    public void M()
+    {
+        void L()
+        {
+            Console.WriteLine(Value);
+        }
+        var f = (Action)(() => L());
+        f();
+    }
+}
+class C2
+{
+    public static void Main(string[] args) => new C<int>().M();
+}
+";
+            VerifyOutput(src, "0");
+        }
+
+        [Fact]
+        public void CaptureGenericParam()
+        {
+            var src = @"
+using System;
+class C<T>
+{
+    T Value = default(T);
+    public void M<U>(U val2)
+    {
+        void L()
+        {
+            Console.WriteLine(Value);
+            Console.WriteLine(val2);
+        }
+        var f = (Action)(() => L());
+        f();
+    }
+}
+class C2
+{
+    public static void Main(string[] args) => new C<int>().M(10);
+}
+";
+            VerifyOutput(src, @"0
+10");
+        }
+
+        [Fact]
+        public void CaptureGenericParamInGenericLocalFunc()
+        {
+            var src = @"
+using System;
+class C<T>
+{
+    T Value = default(T);
+    public void M<U>(U v1)
+    {
+        void L<V>(V v2) where V : T
+        {
+            Console.WriteLine(Value);
+            Console.WriteLine(v1);
+            Console.WriteLine(v2);
+        }
+        var f = (Action)(() => L<T>(Value));
+        f();
+    }
+}
+class C2
+{
+    public static void Main(string[] args) => new C<int>().M(10);
+}
+";
+            VerifyOutput(src, @"0
+10
+0");
+        }
```

Comment:
I see at least one test, `StructClosureGeneric`, in the same file that seems to test that. Was there something specific about the nesting that I should test?